libdevice/itt_stubs.cpp

```diff
@@ -0,0 +1,25 @@
+//==--- itt_stubs.cpp - stub functions for ITT  ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "device_itt.h"
+
+#ifdef __SPIR__
+
+DEVICE_EXTERN_C ITT_STUB_ATTRIBUTES void
```

Comment:
Hi, @vzakhari
Those "stub" functions are meant to be empty or the actual implementation will be added later?
It seems that the "stub" functions are linked with user's device code too, so what's the purpose of splitting the "wrapper" and "stub" into 2 libs?

Reply:
The `stub` functions are meant to be empty.  The idea is that these `stubs` are going to be linked always.  The `compiler wrappers`, on the other hand, may be linked conditionally (e.g. under some compiler option), so it makes sense to separate them.  There is going to be another part with `user wrappers` later.  This one may be combined with the `stubs`, but I hope to move the `user wrappers` into the `ittapi` repo at some point, so I am going to have `user wrappers` in a separate file too.

Reply:
Hi, @vzakhari 
So, all functions in 'stub' code will always be linked with user's device code no matter whether they are used or not and I think we need to support the stub libraries in current "2-step" link mechanism for device libraries. All device libraries are linked using "OnlyNeeded" flag for optimization purpose but we can't link the stub libs using this flag, is my understanding correct?
Thanks very much.

Reply:
This is correct, @jinge90.

The `compiler wrappers` part must be linked always, because calls to these APIs may be introduced as late as in `sycl-post-link`.  If the `sycl-post-link` pass (#3299) uses any of the `user wrappers`, then they will have to be linked always as well.

If any of the wrappers is used by the `sycl-post-link` pass, then the `stub` part must be linked always as well.

Reply:
It looks like we are going to move the instrumentation pass from `sycl-post-link` to the SYCL device code compilation phase, so we should be able to use the same linking scheme for the `ITT` parts as for the rest of `libdevice`.